File: TtBlog/manage/static/js/biz/system.js
var vm = avalon.define({
    $id: 'system',

    name: '',
    summary: '',
    banner: '',
    pageSize: 15,

    csrfmiddlewaretoken: '',

    
    getSetting: function () {
        $.getJSON('/api/getsetting',{
            stamp: Date().toString()
        }, function (res) {
            if(res.result != null)
                vm.name = res.result.Name;
                vm.summary = res.result.Summary;
                vm.banner = res.result.Banner;
        })
    },

    saveSetting: function () {
        if(vm.name == ''){
            layer.alert("请填写必要的信息！");
            return;
        }

        $.post('/api/savesetting', {
            name: vm.name,
            summary: vm.summary,
            banner: vm.banner,
            pageSize: vm.pageSize,
            csrfmiddlewaretoken: vm.csrfmiddlewaretoken
        }, function (res) {
            if(res.result){
                layer.alert("站点配置已保存！")
            } else{
                layer.alert("保存站点配置失败，原因：" + res.err);
            }

        }, "JSON")
    },


})

File: TtBlog/api/system.py
from django.contrib.auth.models import User
from django.views.decorators.csrf import csrf_protect
from django.http import HttpResponse, JsonResponse
import os
import datetime
import json
import re

from blog import models
from TtBlog import settings


def getSetting(request):

    setting = models.Site.objects.all()[0]
    data = {'Id': setting.Id, 'Name': setting.Name, 'Summary': setting.Summary, 'Banner': setting.Banner, 'PageSize': setting.PageSize}

    return JsonResponse({'result': data})


@csrf_protect
def saveSetting(request):

    name = request.POST.get('name')
    summary = request.POST.get('summary')
    banner = request.POST.get('banner')
    pageSize = request.POST.get('pageSize')

    if name is None or name == '':
        return JsonResponse({'result': False, 'err': '调用方法缺少必要的参数！'})

    sites = models.Site.objects.all()
    for s in sites:
        s.delete()

    setting = models.Site()
    setting.Name = name
    setting.Summary = summary
    setting.Banner = banner
    setting.PageSize = pageSize

    setting.save()

    return JsonResponse({'result': True})
